web/approval.go

```diff
@@ -297,15 +311,40 @@ func calculateApprovalInfo(request *model.ApprovalRequest, policy *model.Approva
 	if err != nil {
 		return nil, err
 	}
-	approved = approved && audit && authorAffirm
+	if !approved {
```

Comment:
I'm not sure what the behavior should be. This would be my guess at the algorithm:

1. Check for approval, ignoring author approval rules
2. If not approved, it's not approved
3. If it is approved, check to see if author approval is required
4. If it is, check to see if the approved PR would still be approved with the committers removed from the approval list.
5. If it is, life is good
6. If it is not, check to see if there was an author approval.
7. If there is, life is good.
8. If there is not, approval denied, send a message indicating that either author approval is required or more approvals from non-committers to the PR.

This approach limits messages about author approval until we get to the point where the PR looks like it should accepted, but it isn't. Is that the goal?